chunker.go

```diff
@@ -209,7 +210,8 @@ func (c *Chunker) split(i int, err error) (uint64, []byte, error) {
 	// reset the hash
 	c.hIdx = 0
 	c.hValue = 0
-	return start, b, err
+	id := sha512.Sum512_256(b)
```

Comment:
Calculating the checksum here, rather than in the worker processes is going to make the chunking process much slower since this is done in the main goroutine.

Reply:
Oh boy. Silly me, I got overly excited with the encapsulation and totally overlooked this. Reverting this in next rev.